replica/table.cc

```diff
@@ -1736,20 +1736,23 @@ future<db::replay_position> table::discard_sstables(db_clock::time_point truncat
             cf.refresh_compound_sstable_set();
         }
     };
-        // FIXME: indentation.
-        auto p = make_lw_shared<pruner>(*this, *_compaction_group);
+    db::replay_position rp;
+    co_await for_each_compaction_group([this, truncated_at, &rp] (compaction_group& cg) -> future<> {
+        auto p = make_lw_shared<pruner>(*this, cg);
```

Comment:
We should consider doing this in parallel to avoid delaying the commit log. Not critical.